manager/src/main/java/org/openremote/manager/security/KeyStoreServiceImpl.java

```diff
@@ -0,0 +1,200 @@
+package org.openremote.manager.security;
+
+import org.openremote.agent.protocol.mqtt.CustomKeyManagerFactory;
+import org.openremote.agent.protocol.mqtt.CustomX509TrustManagerFactory;
+import org.openremote.container.persistence.PersistenceService;
+import org.openremote.model.Container;
+import org.openremote.model.security.KeyStoreService;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManagerFactory;
+import java.io.*;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.util.logging.Logger;
+import java.util.Optional;
+
+import static org.openremote.container.util.MapAccess.getString;
+
+/**
+ * <p>
+ * This service is used for retrieving, creating, or editing KeyStore (and TrustStore) files.
+ * </p><p>
+ * Currently the KeyStores are stored in the Storage directory of OpenRemote, which is usually volumed and persisted.
+ * </p><p>
+ * Each realm is allocated 2 KeyStores; A Client KeyStore, which is used for storing key-pairs that are used by
+ * clients (be that as an Agent or as a plain client), and a TrustStore, which contains trusted certificates, usually SSL
+ * certificates. To ensure both security and extensibility, both the default and predefined TrustStores are used to find
+ * the correct certificates.
+ * </p>
+ */
+public class KeyStoreServiceImpl implements KeyStoreService {
+
+    protected PersistenceService persistenceService;
+    protected ManagerIdentityService identityService;
+
+    private KeyStore keyStore = null;
+    private KeyStore trustStore = null;
+
+    private final String OR_SSL_CLIENT_KEYSTORE_FILE = "OR_SSL_CLIENT_KEYSTORE_FILE";
+    private final String OR_SSL_CLIENT_TRUSTSTORE_FILE = "OR_SSL_CLIENT_TRUSTSTORE_FILE";
+
+    private final String OR_SSL_CLIENT_KEYSTORE_PASSWORD = "OR_SSL_CLIENT_KEYSTORE_PASSWORD";
+    private final String OR_SSL_CLIENT_TRUSTSTORE_PASSWORD = "OR_SSL_CLIENT_TRUSTSTORE_PASSWORD";
+
+    private final String OR_KEYSTORE_PASSWORD = "OR_KEYSTORE_PASSWORD";
+
+
+    private Path keyStorePath;
+    private Path trustStorePath;
```

Comment:
prefer protected as it allows extensions of the service to be created more easily